Wave Generator crashes on orientation change

**Actual Behaviour**

The Wave Generator crashes on orientation change.

**Expected Behaviour**

The app should never crash when the orientation of the device is changed.

1. Open PSLab app
2. Go to Wave Generator
3. Tap "Frequency" button
4. Change frequency via seek bar
5. Change orientation of the device

**LogCat for the issue**
`FATAL EXCEPTION: main
Process: io.pslab, PID: 7816
java.lang.RuntimeException: Unable to start activity ComponentInfo{io.pslab/io.pslab.activity.WaveGeneratorActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.hashCode()' on a null object reference
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2678)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2743)
	at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4560)
	at android.app.ActivityThread.-wrap19(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1496)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6165)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:888)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:778)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.hashCode()' on a null object reference
	at io.pslab.activity.WaveGeneratorActivity.setValue(WaveGeneratorActivity.java:945)
	at io.pslab.activity.WaveGeneratorActivity.-$$Nest$msetValue(WaveGeneratorActivity.java)
	at io.pslab.activity.WaveGeneratorActivity$16.onSeeking(WaveGeneratorActivity.java:432)
	at com.warkiz.widget.IndicatorSeekBar.setSeekListener(IndicatorSeekBar.java:1434)
	at com.warkiz.widget.IndicatorSeekBar.setProgress(IndicatorSeekBar.java:1634)
	at com.warkiz.widget.IndicatorSeekBar.onRestoreInstanceState(IndicatorSeekBar.java:1185)
	at android.view.View.dispatchRestoreInstanceState(View.java:15767)
	at android.view.ViewGroup.dispatchRestoreInstanceState(ViewGroup.java:3240)
	at android.view.ViewGroup.dispatchRestoreInstanceState(ViewGroup.java:3240)
	at android.view.ViewGroup.dispatchRestoreInstanceState(ViewGroup.java:3240)
	at android.view.ViewGroup.dispatchRestoreInstanceState(ViewGroup.java:3240)
	at android.view.ViewGroup.dispatchRestoreInstanceState(ViewGroup.java:3240)
	at android.view.ViewGroup.dispatchRestoreInstanceState(ViewGroup.java:3240)
	at android.view.ViewGroup.dispatchRestoreInstanceState(ViewGroup.java:3240)
	at android.view.View.restoreHierarchyState(View.java:15745)
	at com.android.internal.policy.PhoneWindow.restoreHierarchyState(PhoneWindow.java:2106)
	at android.app.Activity.onRestoreInstanceState(Activity.java:1051)
	at android.app.Activity.performRestoreInstanceState(Activity.java:1006)
	at android.app.Instrumentation.callActivityOnRestoreInstanceState(Instrumentation.java:1196)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2651)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2743) 
	at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:4560) 
	at android.app.ActivityThread.-wrap19(ActivityThread.java) 
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1496) 
	at android.os.Handler.dispatchMessage(Handler.java:102) 
	at android.os.Looper.loop(Looper.java:154) 
	at android.app.ActivityThread.main(ActivityThread.java:6165) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:888) 
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:778) 

**Would you like to work on the issue?**
Maybe.

--- a/app/src/main/java/io/pslab/activity/WaveGeneratorActivity.java
+++ b/app/src/main/java/io/pslab/activity/WaveGeneratorActivity.java
@@ -174,7 +174,7 @@ public class WaveGeneratorActivity extends GuideActivity {
     private AlertDialog waveDialog;
     private CSVLogger csvLogger;
     private WaveConst waveBtnActive, pwmBtnActive, prop_active, digital_mode;
-    private TextView activePropTv = null;
+    private TextView activePropTv;
     private CoordinatorLayout coordinatorLayout;
     private Realm realm;
     private GPSLogger gpsLogger;
@@ -417,12 +417,7 @@ public class WaveGeneratorActivity extends GuideActivity {
         seekBar.setOnSeekChangeListener(new OnSeekChangeListener() {
             @Override
             public void onSeeking(SeekParams seekParams) {
-                String valueText;
-                if ("\u00b0".equals(unit)) {
-                    valueText = seekParams.progress + unit;
-                } else {
-                    valueText = seekParams.progress + " " + unit;
-                }
+                String valueText = formatWithUnit(seekParams.progress, unit);
 
                 if (waveMonSelected) {
                     waveMonPropValueSelect.setText(valueText);
@@ -470,6 +465,8 @@ public class WaveGeneratorActivity extends GuideActivity {
         if (getResources().getBoolean(R.bool.isTablet)) {
             setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_USER);
         }
+
+        activePropTv = prop_active == WaveConst.FREQUENCY ? waveFreqValue : wavePhaseValue;
     }
 
     public void saveWaveConfig() {
@@ -678,11 +675,6 @@ public class WaveGeneratorActivity extends GuideActivity {
         return true;
     }
 
-    @Override
-    protected void onResume() {
-        super.onResume();
-    }
-
     public void selectBtn(WaveConst btn_selected) {
 
         switch (btn_selected) {
@@ -835,16 +827,8 @@ public class WaveGeneratorActivity extends GuideActivity {
     }
 
     private void fetchPropertyValue(WaveConst btnActive, WaveConst property, String unit, TextView propTextView) {
-        Double value = (double) WaveGeneratorConstants.wave.get(btnActive).get(property);
-        String valueText;
-        switch (unit) {
-            case "\u00b0":
-                valueText = value.intValue() + unit;
-                break;
-            default:
-                valueText = value.intValue() + " " + unit;
-        }
-        propTextView.setText(valueText);
+        int value = WaveGeneratorConstants.wave.get(btnActive).get(property);
+        propTextView.setText(formatWithUnit(value, unit));
     }
 
     private void setSeekBar(IndicatorSeekBar seekBar) {
@@ -912,7 +896,7 @@ public class WaveGeneratorActivity extends GuideActivity {
     }
 
     private void decProgressSeekBar(IndicatorSeekBar seekBar) {
-        Integer value = seekBar.getProgress();
+        int value = seekBar.getProgress();
         value = value - leastCount;
         if (value < seekMin) {
             value = seekMin;
@@ -921,7 +905,7 @@ public class WaveGeneratorActivity extends GuideActivity {
     }
 
     private void setValue() {
-        Integer value = seekBar.getProgress();
+        int value = seekBar.getProgress();
 
         if (!waveMonSelected) {
             if (prop_active == WaveConst.FREQUENCY) {
@@ -940,16 +924,11 @@ public class WaveGeneratorActivity extends GuideActivity {
         }
         setWave();
         previewWave();
-        Double dValue = (double) value;
-        String valueText;
-        switch (unit) {
-            case "\u00b0":
-                valueText = dValue.intValue() + unit;
-                break;
-            default:
-                valueText = dValue.intValue() + " " + unit;
-        }
-        activePropTv.setText(valueText);
+        activePropTv.setText(formatWithUnit(value, unit));
+    }
+
+    private static String formatWithUnit(int value, String unit) {
+        return value + ("\u00b0".equals(unit) ? "" : " ") + unit;
     }
 
     private void previewWave() {

